app/models/cart.rb

```diff
@@ -46,11 +46,15 @@ class Cart < FlexCommerceApi::ApiBase
     has_one :billing_address, class_name: "::FlexCommerce::Address"
     has_one :shipping_method, class_name: "::FlexCommerce::ShippingMethod"
     has_one :free_shipping_promotion, class_name: "::FlexCommerce::FreeShippingPromotion"
+    
+    # properties
+    property :line_items_count, type: :integer, default: 0
+    property :total_discount, default: 0
 
     # Here we override line_items to provide a proxy to the array so we can use new and create on it in the normal
     # active record way
     def line_items
-      has_many_association_proxy :line_items, super, inverse_of: :container
+      has_many_association_proxy :line_items, (super || []), inverse_of: :container
```

Comment:
This prevents an error when doing things like `FlexCommerce::Cart.new.line_items.any?` (as `any?` would be called on `nil`, so this way we pass in a blank array as a default)